q2_diversity_lib/tests/test_beta.py

```diff
@@ -321,10 +322,152 @@ def test_accepted_types_have_consistent_behavior(self):
+        super().setUp()
+        self.method = self.plugin.actions['beta_passthrough']
+        self.available_metrics = beta.METRICS['NONPHYLO']['UNIMPL']
+        empty_table = biom.Table(np.array([]), [], [])
+        self.empty_table = Artifact.import_data('FeatureTable[Frequency]',
+                                                empty_table)
+        crawford_tbl = self.get_data_path('crawford.biom')
+        self.crawford_tbl = Artifact.import_data('FeatureTable[Frequency]',
+                                                 crawford_tbl)
+        table = biom.Table(np.array([[0, 1, 3], [1, 1, 2]]),
+                           ['O1', 'O2'],
+                           ['S1', 'S2', 'S3'])
+        self.table = Artifact.import_data('FeatureTable[Frequency]', table)
+
+    def testMethod(self):
+        for metric in self.available_metrics:
+            self.method(table=self.crawford_tbl, metric=metric)
+        # If we get here, then our methods ran without error
+        self.assertTrue(True)
+
+    def test_passed_empty_table(self):
+        for metric in self.available_metrics:
+            with self.assertRaisesRegex(ValueError, 'empty'):
+                self.method(table=self.empty_table, metric=metric)
+
+    def test_passed_bad_metric(self):
+        with self.assertRaisesRegex(TypeError,
+                                    'imaginary_metric.*incompatible'):
+            self.method(table=self.crawford_tbl, metric='imaginary_metric')
+
+    def test_passed_implemented_metric(self):
+        # beta_passthrough does not provide access to measures that have been
+        # implemented locally
+        for metric in beta.METRICS['NONPHYLO']['IMPL']:
+            with self.assertRaisesRegex(TypeError, f"{metric}.*incompatible"):
+                self.method(table=self.crawford_tbl, metric=metric)
+
+    def test_aitchison(self):
+        actual = self.method(table=self.table, metric='aitchison')
+        actual = actual[0].view(skbio.DistanceMatrix)
+        expected = skbio.DistanceMatrix([[0.0000000, 0.4901290, 0.6935510],
+                                         [0.4901290, 0.0000000, 0.2034219],
+                                         [0.6935510, 0.2034219, 0.0000000]],
+                                        ids=['S1', 'S2', 'S3'])
+
+        self.assertEqual(actual.ids, expected.ids)
+        for id1 in actual.ids:
+            for id2 in actual.ids:
+                npt.assert_almost_equal(actual[id1, id2], expected[id1, id2])
+
+    def test_canberra_adkins(self):
+        t = biom.Table(np.array([[0, 0], [0, 1], [1, 2]]),
+                       ['O1', 'O2', 'O3'],
+                       ['S1', 'S2'])
+        t = Artifact.import_data('FeatureTable[Frequency]', t)
+        d = (1. / 2.) * sum([abs(0. - 1.) / (0. + 1.),
+                             abs(1. - 2.) / (1. + 2.)])
+        expected = skbio.DistanceMatrix(np.array([[0.0, d], [d, 0.0]]),
+                                        ids=['S1', 'S2'])
```

Comment:
Let's not re-implement the metric logic here - instead, please just fill in the expected values in the `expected` DM (like in the jensenshannon test below).